controllers/infinispan_controller.go

```diff
@@ -1762,18 +1765,36 @@ func GossipRouterPodList(infinispan *infinispanv1.Infinispan, kube *kube.Kuberne
 	return podList, kube.ResourcesList(infinispan.Namespace, GossipRouterPodLabels(infinispan.Name), podList, ctx)
 }
 
-func buildStartupArgs(overlayConfigMapKey string, zeroCapacity string) []string {
-	var args []string
-	if overlayConfigMapKey != "" {
-		args = []string{"-Dinfinispan.zero-capacity-node=" + zeroCapacity, "-l", OperatorConfMountPath + "/log4j.xml", "-c", "user/" + overlayConfigMapKey, "-c", "operator/infinispan.xml"}
+func buildStartupArgs(overlayConfigMapKey, overlayLog4jConfigMapKey, zeroCapacity string) []string {
+	var args strings.Builder
+
+	// Preallocate a buffer to speed up string building (saves code from growing the memory dynamically)
+	args.Grow(110)
+	args.WriteString("-Dinfinispan.zero-capacity-node=")
+	args.WriteString(zeroCapacity)
+
+	// Check if the user defined a custom log4j config
+	args.WriteString(" -l ")
+	if overlayLog4jConfigMapKey != "" {
+		args.WriteString("user/log4j.xml")
 	} else {
-		args = []string{"-Dinfinispan.zero-capacity-node=" + zeroCapacity, "-l", OperatorConfMountPath + "/log4j.xml", "-c", "operator/infinispan.xml"}
+		args.WriteString(OperatorConfMountPath)
+		args.WriteString("/log4j.xml")
 	}
-	return args
+
+	// Check if the user defined an overlay operator config
+	args.WriteString(" -c ")
+	if overlayConfigMapKey != "" {
+		args.WriteString("user/")
+		args.WriteString(overlayConfigMapKey)
+	}
+	args.WriteString("operator/infinispan.xml")
```

Comment:
We still need to provide the " -c " here, so `" -c operator/infinispan.xml"`

Reply:
done, thanks for pointing it out.

Reply:
Had to also add the overlay `log4j.xml` snipped to the `ConfigMaps` on both tests `TestUserCustomConfigUpdateOnNameChange` and `TestUserCustomConfigWithAuthUpdate`.